### PR TITLE
[Dice-206]: support `KEEPTTL` option for `SET` command

### LINKED ISSUE #206: Add support for `KEEPTTL` in `SET` command
We already have the `SET` command implemented. Now we need to add support for `KEEPTTL` option similar to the [`KEEPTTL` option in Redis](https://redis.io/docs/latest/commands/set/). Please refer to the following commit in Redis to understand the implementation specifics - [source](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa).

Write unit and integration tests for the command referring to the tests written in the [Redis codebase 7.2.5](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa). For integration tests, you can refer to the `tests` folder. Note: they have used `TCL` for the test suite, and we need to port that to our way of writing integration tests using the relevant helper methods. Please refer to our [tests directory](https://github.com/DiceDB/dice/tree/master/tests).

For the command, benchmark the code and measure the time taken and memory allocs using [`benchmem`](https://blog.logrocket.com/benchmarking-golang-improve-function-performance/) and try to keep them to the bare minimum.

### PR DESCRIPTION
**Pull request by kolharsam**
this is a newer version of the PR #212 that resolves #206 

@JyotinderSingh please review this instead of the older PR. I am closing the older one since there were too many conflicts to deal with. I hope that this is ok!

### CODE REVIEW

File: core/session_test.go [hunk 1]
```diff
@@ -104,10 +104,8 @@ func TestSessionActivate(t *testing.T) {
 	session := NewSession()
 	user := &User{Username: DefaultUserName}
 
-	err := session.Activate(user)
-	if err != nil {
-		t.Errorf("Session.Activate() returned an error: %v", err)
-	}
+	session.Activate(user)
+
```
Comment (JyotinderSingh) [inline core/session_test.go:108]: Why is this change needed?

File: core/store.go [hunk 2]
```diff
@@ -58,18 +58,42 @@ func NewObj(value interface{}, expDurationMs int64, oType, oEnc uint8) *Obj {
...
+	KeepTTL bool
+}
+
+func Put(k string, obj *Obj, opts ...PutOption) {
+	options := getDefaultOptions()
+
+	for _, optApplier := range opts {
+		optApplier(options)
+	}
+
```
Comment (JyotinderSingh) [inline core/store.go:71]: Do you think it will be better if we do this inside the putHelper (and simply pass the opts directly to it)?

That way we can probably get rid of the nil params being passed to putHelper in a few other places. Additionally, we won't need to do an `opts != nil` check inside the helper.

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for these changes @kolharsam! Left some minor comments.

File: core/session_test.go (hunk 1 above)
Comment (kolharsam) [reply core/session_test.go:108]: image

My editor was complaining about this issue and so I made this change. I understand it is not related to the PR, would like me to remove it?

File: core/store.go (hunk 2 above)
Comment (kolharsam) [reply core/store.go]: Thanks for the note @JyotinderSingh.
I have made the changes for the same.

**Comment by kolharsam** [comment]
> Please resolve the linter issues too.

I have resolved these too @JyotinderSingh, hope that it's all okay! 

Thanks!

**Review by JyotinderSingh — APPROVED** [review]
Thanks for making the changed, approved.
